### PR TITLE
add missing fields to SOA

### LINKED ISSUE #1282: Can only edit slave zone name, group, master zones, and allowed AXFR ips
screen shot 2017-02-24 at 1 12 40 pm

### LINKED ISSUE #1295: SOA record is missing domain transfer (AXFR) section

### PR DESCRIPTION
**Pull request by na3d**
Closes #1282
Closes #1295 
screen shot 2017-05-12 at 4 57 21 pm
screen shot 2017-05-12 at 4 57 28 pm

### CODE REVIEW

File: components/scss/components/forms/form.scss [hunk 1]
```diff
@@ -81,6 +81,12 @@ textarea {
     }
 }
 
+.Modal {
+    textarea {
+        min-width: 100%;
+    }
+}
+
```
Comment (eatonphil) [inline components/scss/components/forms/form.scss]: Please just add textarea to the list of .Input and .Select here instead:

https://github.com/linode/manager/blob/master/components/scss/components/modals/Modal.scss#L55

Comment (na3d) [inline components/scss/components/forms/form.scss]: Ah!  Thank you!

File: components/scss/components/modals/Modal.scss [hunk 2]
```diff
@@ -59,6 +59,10 @@
             width: 100%;
         }
     }
+
+    textarea {
+        min-width: 100%;
+    }
```
Comment (eatonphil) [inline components/scss/components/modals/Modal.scss]: Actually you can get rid of this now. If you rebase on master its already in there: https://github.com/linode/manager/blob/master/components/scss/components/modals/Modal.scss#L55

Comment (na3d) [inline components/scss/components/modals/Modal.scss]: It isn't the same.
`min-width`
Needs to be overwrote.

File: src/domains/components/EditSOARecord.js [hunk 3]
```diff
@@ -84,85 +95,144 @@ export default class EditSOARecord extends Component {
...
+              label="Expire Time"
+              description="How long secondary (slave) nameservers wait
+                before considering DNS data stale if it cannot reach the
+                master nameserver"
+              id="expireTime"
+              apiKey="expire_sec"
+              errors={errors}
+            >
+              <Select
+                value={Math.max(expireTime, 604800).toString()}
```
Comment (mparke) [inline src/domains/components/EditSOARecord.js]: these numeric values should be constants

Comment (mparke) [inline src/domains/components/EditSOARecord.js]: here and in options

Comment (eatonphil) [inline src/domains/components/EditSOARecord.js]: There is a Select component for editing these times. It should be using that.

Comment (eatonphil) [inline src/domains/components/EditSOARecord.js]: Oh wait nevermind, you can't use that here.

Comment (eatonphil) [inline src/domains/components/EditSOARecord.js]: @na3d also didn't write this code, @mparke , I did. But there are lots of nice juicy constants here if you'd like: https://github.com/linode/manager/blob/master/src/domains/components/SelectDNSSeconds.js#L16

File: src/domains/components/EditSOARecord.js [hunk 4]
```diff
@@ -11,60 +11,71 @@ import SelectDNSSeconds from './SelectDNSSeconds';
...
       group,
-      zone,
+      domain,
       defaultTTL,
       refreshRate,
       retryRate,
       expireTime,
       email,
+      status,
+      axfrIps: axfrIps.join(';'),
```
Comment (mparke) [inline src/domains/components/EditSOARecord.js]: the description below says semicolons or newlines, do we need to account for both here?

Comment (eatonphil) [inline src/domains/components/EditSOARecord.js]: Just pick one please.

Comment (na3d) [inline src/domains/components/EditSOARecord.js]: Remove the other from the description then, I am assuming.

File: src/domains/components/EditSOARecord.js [hunk 5]
```diff
@@ -6,65 +6,80 @@ import { CancelButton } from 'linode-components/buttons';
...
-      domain: zone,
+      group,
+      domain,
       ttl_sec: defaultTTL,
       refresh_sec: refreshRate,
       retry_sec: retryRate,
       expire_sec: expireTime,
       soa_email: email,
+      master_ips: masterIps.length ? masterIps.split(';') : [],
+      axfr_ips: axfrIps.length ? axfrIps.split(';') : [],
```
Comment (eatonphil) [inline src/domains/components/EditSOARecord.js]: The API returns an array for both master_ips and axfr_ips not a string.

Comment (eatonphil) [inline src/domains/components/EditSOARecord.js]: Oh duh, nvm.

**Pull request merged**